client/reducers/recipes/recipesReducer.js

```diff
@@ -0,0 +1,105 @@
+import { mapKeys } from 'lodash';
+
+import {
+  RECEIVE_RECIPE,
+  UPVOTE_RECIPE, ADD_REVIEW,
+  GET_DOWNVOTE,
+  ADD_FAVORITE_SUCCESS,
+  FAVORITE_ID,
+  ADD_RECIPE,
+  ADD_RECIPE_ERROR,
+  UPDATE_RECIPE,
+  UPDATE_RECIPE_ERROR,
+  GET_ALL_RECIPES,
+  GET_MY_RECIPES,
+  DELETE_RECEIPE
+} from '../../action/types';
+
+export const recipesReducer = (state = {}, action) => {
+  switch (action.type) {
+    case GET_ALL_RECIPES:
+      return {
+        ...state,
+        recipes: action.recipes
+      };
+
+    case GET_MY_RECIPES:
+      return {
+        ...state,
+        recipes: action.recipes
+      };
+
+    case RECEIVE_RECIPE:
+      return {
+        ...action.recipe,
+        Reviews: mapKeys(action.recipe.Reviews, 'id')
+      };
+
+    case UPVOTE_RECIPE:
+      return {
+        ...state,
+        upVotes: action.recipe.upVotes,
+        downVotes: action.recipe.downVotes,
+      };
+
+    case ADD_RECIPE:
+      return {
+        ...state,
+        recipe: action.recipe
+      };
+
+    case ADD_RECIPE_ERROR:
+      return {
+        ...state,
+        serverError: action.error
+      };
+
+    case UPDATE_RECIPE:
+      return {
+        ...state,
+        recipe: action.recipe
+      };
+
+    case UPDATE_RECIPE_ERROR:
+      return {
+        ...state,
+        serverError: action.error
+      };
+
+    case ADD_FAVORITE_SUCCESS:
+      return {
+        ...state,
+        favoriteCheck: action.favoriteCheck
+      };
+
+    case FAVORITE_ID:
+      return {
+        ...state,
+        favoritesId: action.favoritesId
+      };
+
+    case GET_DOWNVOTE:
+      return {
+        ...state,
+        action: action.recipe
+      };
+
+    case ADD_REVIEW:
+      return {
+        ...state,
+        ...state.recipe,
+        Reviews: mapKeys(action.recipe.Reviews, 'id')
+      };
+
+    case DELETE_RECEIPE: {
+      const deletedRecipe = state.myRecipes
+        .filter(recipe => (recipe.id !== action.recipes));
+      return {
+        ...state,
+        recipes: deletedRecipe,
+      };
+    }
+    default:
+      return state;
+  }
+};
```

Comment:
Nitpick:
When you have very large reducers, you can split them into smaller reducers, then use combineReducers to combine them into a single reducer